File: sdk/java/src/main/java/io/ancora/AncorException.java
package io.ancora;

import io.ancora.ffi.AncorErrorCode;

/** Exception raised when an ancora-ffi function returns a non-OK error code. */
public final class AncorException extends Exception {

    private final int errorCode;

    public AncorException(int errorCode, String message) {
        super(message + " (ErrorCode=" + errorCode + ":" + AncorErrorCode.from(errorCode) + ")");
        this.errorCode = errorCode;
    }

    public AncorException(AncorErrorCode code, String message) {
        this(code.value, message);
    }

    public int getErrorCode() {
        return errorCode;
    }
}

File: sdk/dotnet/Ancora/AncorException.cs
using System;

namespace Ancora;

/// <summary>
/// Exception raised when an Ancora FFI function returns a non-Ok error code.
/// </summary>
public sealed class AncorException : Exception
{
    /// <summary>
    /// The integer error code from the FFI call.
    /// </summary>
    public int ErrorCode { get; }

    public AncorException(int errorCode, string message)
        : base($"{message} (ErrorCode={errorCode})")
    {
        ErrorCode = errorCode;
    }
}
